src/sentry/tasks/integrations.py

```diff
@@ -19,37 +19,85 @@
 # TODO(jess): Add more retry exclusions once ApiClients have better error handling
 @retry(exclude=(ExternalIssue.DoesNotExist, Integration.DoesNotExist))
-def post_comment(external_issue_id, data, user_id, **kwargs):
-    # sync Sentry comments to an external issue
+def create_comment(external_issue_id, user_id, group_note_id, **kwargs):
     external_issue = ExternalIssue.objects.get(id=external_issue_id)
+    installation = Integration.objects.get(
+        id=external_issue.integration_id
+    ).get_installation(
+        organization_id=external_issue.organization_id,
+    )
 
-    organization = Organization.objects.get(id=external_issue.organization_id)
-    has_issue_sync = features.has('organizations:integrations-issue-sync',
-                                  organization)
-    if not has_issue_sync:
+    if not should_comment_sync(installation, external_issue):
         return
+    try:
+        note = Activity.objects.get(
+            type=Activity.NOTE,
+            id=group_note_id,
+        )
+    except Activity.DoesNotExist:
+        return
+    comment = installation.create_comment(external_issue.key, user_id, note)
+    note.data['external_id'] = installation.get_comment_id(comment)
+    note.save()
+    analytics.record(
+        # TODO(lb): this should be changed and/or specified?
+        'integration.issue.comments.synced',
+        provider=installation.model.provider,
+        id=installation.model.id,
+        organization_id=external_issue.organization_id,
+        user_id=user_id,
+    )
 
-    integration = Integration.objects.get(id=external_issue.integration_id)
-    installation = integration.get_installation(
+
+@instrumented_task(
+    name='sentry.tasks.integrations.update_comment',
+    queue='integrations',
+    default_retry_delay=60 * 5,
+    max_retries=5
+)
+# TODO(jess): Add more retry exclusions once ApiClients have better error handling
+@retry(exclude=(ExternalIssue.DoesNotExist, Integration.DoesNotExist))
+def update_comment(external_issue_id, user_id, group_note_id, **kwargs):
+    external_issue = ExternalIssue.objects.get(id=external_issue_id)
+    installation = Integration.objects.get(
+        id=external_issue.integration_id
+    ).get_installation(
         organization_id=external_issue.organization_id,
     )
-    if installation.should_sync('comment'):
-        installation.create_comment(external_issue.key, user_id, data['text'])
-        analytics.record(
-            'integration.issue.comments.synced',
-            provider=integration.provider,
-            id=integration.id,
-            organization_id=external_issue.organization_id,
-            user_id=user_id,
+
+    if not should_comment_sync(installation, external_issue):
+        return
+    try:
+        note = Activity.objects.get(
+            type=Activity.NOTE,
+            id=group_note_id,
         )
+    except Activity.DoesNotExist:
+        return
+    installation.update_comment(external_issue.key, user_id, note)
```

Comment:
Because you're in a queue task you could bump into race-like conditions where the comment has been deleted inside jira before your update is dequeued and executed. I think right now that scenario would cause an error.

Reply:
Dispite being created by the Sentry user, I can edit and save my edits over the comment. Why Jira works like that I don't know. The comment is still attributed to the Sentry user. But my edits persisted. So we do in fact hit the issue @markstory  highlighted. I'm putting this on hold for now until I can ask @saragilford how product would like to proceed. I believe this is lower priority, so it should be ok.

Reply:
<img width="927" alt="screen shot 2019-01-22 at 2 40 58 pm" src="https://user-images.githubusercontent.com/12959365/51570539-1df21000-1e54-11e9-8ecb-ac7fab343601.png">

Reply:
@markstory I spoke with @saragilford and she decided that she wanted to continue ahead with this change disregarding the issues we discussed around this.

Reply:
To elaborate more on the discussion @lauryndbrown and I had. To my understanding:
- this would affect users who 1) have issue sync 2) edit the same comment twice -- once in Jira and then again in Sentry
- the way to prevent overwriting of the comment in this case would be to build out two-way sync for comments, which we de-prioritized early on in the process.
To my mind, the above situation is a corner case, and not worth building out two way comment sync for Jira (and for parity, Azure Devops) now. If there is a simpler solution, we might consider moving forward, but I would put this case as pretty low-priority.

Reply:
@saragilford just a quick note: Adding this feature breaks parity with Azure DevOps because you cannot edit comments in Azure DevOps. So Jira will be the only one with this feature.